solidity/dashboard/src/sagas/rewards.js

```diff
@@ -68,12 +75,40 @@ export function* watchWithdrawECDSARewards() {
 }
 
 function* fetchECDSARewardsData(action) {
-  const beneficiary = action.payload
+  const address = action.payload
   try {
     yield put({ type: "rewards/ecdsa_fetch_rewards_data_start" })
 
-    const operators = yield call(getOperatorsOfBeneficiary, beneficiary)
-    let availableRewards = yield call(fetchECDSAAvailableRewards, operators)
+    // Beneficiary operators.
+    const beneficiaryOperators = yield call(getOperatorsOfBeneficiary, address)
+    // Owner operators.
+    const ownerOperators = yield call(getOperatorsOfOwner, address)
+    // Grantee operators.
+    const grenteeOperators = yield call(getOperatorsOfGrantee, address)
+    // Managed grantee operators.
+    const mangedGranteeOperators = yield call(
+      getOperatorsOfManagedGrantee,
+      address
+    )
+    // Get operators of copied delegations where an owner of the old delegations
+    // is `address`.
+    const copiedOperators = yield call(getOperatorsOfCopiedDelegations, address)
+    // The same address can be used as beneficiary and owner. So addresses in
+    // array may be repeated. Let's convert to a Set to make sure the given
+    // address is in array only once.
+    const operators = new Set(
+      beneficiaryOperators
+        .concat(ownerOperators)
+        .concat(grenteeOperators)
+        .concat(mangedGranteeOperators)
+        .concat(copiedOperators)
+    )
```

Comment:
I think the UI is not user-friendly when we try to withdraw all tbtc rewards for more than one reward. For example when we have 5 rows in rewards history section

![image](https://user-images.githubusercontent.com/40306834/108865401-81326b80-75f3-11eb-8a02-158e45ed0ddd.png)

and we click **Withdraw All** button we have to do 5 MetaMask transactions on after another which might be a little problematic for higher number of rewards. I'm not sure how we can improve that tho 🤔 Any thoughts?

Reply:
There is also another issue with this. When, after clicking the **Withdraw All** button, I accept the first MetaMask transaction and then reject the second one I can't withdraw the rest of the rewards anymore and MetaMask shows a transaction error. After refresh it shows me that all the rewards are **Available**, which should not be the case because i withdrew the first one.

Reply:
> I think the UI is not user-friendly when we try to withdraw all tbtc rewards for more than one reward. For example when we have 5 rows in rewards history section
> 
> ![image](https://user-images.githubusercontent.com/40306834/108865401-81326b80-75f3-11eb-8a02-158e45ed0ddd.png)
> 
> and we click **Withdraw All** button we have to do 5 MetaMask transactions on after another which might be a little problematic for higher number of rewards. I'm not sure how we can improve that tho 🤔 Any thoughts?

We are limited by [`ECDSARewardsDistributor`](https://github.com/keep-network/keep-ecdsa/blob/master/solidity/contracts/ECDSARewardsDistributor.sol#L72) contract. We can't withdraw rewards for a given number of operators in a single transaction. Another solution is to add `withdraw` button per row.